### PR TITLE
*: cherry pick some commits to release-2.0

### PR DESCRIPTION
**Pull request by huachaohuang**
## What have you changed? (mandatory)

Cherry pick some commits to release-2.0.

## What are the type of the changes? (mandatory)

- Cherry pick

## Refer to a related PR or issue link (optional)

https://github.com/pingcap/kvproto/pull/276

### CODE REVIEW

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -16,6 +16,7 @@
 #![cfg_attr(not(feature = "dev"), allow(unknown_lints))]
 #![allow(needless_pass_by_value)]
 
+#[macro_use]
```
Comment (overvenus) [inline src/bin/tikv-ctl.rs]: Does it necessary?

Comment (huachaohuang) [reply src/bin/tikv-ctl.rs]: Yes, it is added in another PR, but that is not picked to release-2.0

**Comment by disksing** [comment]
LGTM

**Comment by siddontang** [comment]
/run-all-tests tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by huachaohuang** [comment]
/run-unit-test tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by huachaohuang** [comment]
/run-all-tests tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by huachaohuang** [comment]
/run-unit-test tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by huachaohuang** [comment]
/run-unit-test tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by huachaohuang** [comment]
/run-all-tests tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by huachaohuang** [comment]
/run-all-tests tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by huachaohuang** [comment]
/run-all-tests tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by siddontang** [comment]
/run-all-tests tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by huachaohuang** [comment]
/run-all-tests tidb=release-2.0 tikv=release-2.0 tidb-test=release-2.0

**Comment by DorianZheng** [comment]
I think I need to do this for my PR too, pls tell me how this work and how
to trigger this?

2018-06-16 0:40 GMT+08:00 Huachao Huang <notifications@github.com>:

**Comment by siddontang** [comment]
/run-all-tests tidb=release-2.0 pd=release-2.0 tidb-test=release-2.0

**Comment by huachaohuang** [comment]
@siddontang Why you run all tests again?

**Comment by siddontang** [comment]
@huachaohuang 

you must pass pd=release-2.0, not tikv=release-2.0

**Comment by huachaohuang** [comment]
/run-all-tests tidb=release-2.0 pd=release-2.0 tidb-test=release-2.0

**Pull request merged**